fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -457,18 +459,23 @@ public void close() {
      */
     @Nonnull
     public CompletableFuture<Void> buildRange(@Nullable Key.Evaluated start, @Nullable Key.Evaluated end) {
-        return recordStoreBuilder.getSubspaceProvider().getSubspaceAsync().thenCompose(subspace -> buildRange(subspace, start, end));
+        return buildRange(recordStoreBuilder.getSubspaceProvider(), start, end);
```

Comment:
The builder could have a `getSubpaceAsync` method that had those same checks.

Reply:
The current builder logic will not complain if you set do setSubspace, setSubspaceProvider , setKeySpacePath, over and over. It will quietly set SubspaceProvider to the last of those setters to execute. Moreover, the first of those two could provide a subspace that has been already "resolved" to a database, which can be inconsistent with whatever context you ultimately .build() with. This PR does not change any of that. Ideally, when build() is done there should be a way to check the consistency of all of this relative to whatever context is used for the record store.

Reply:
No changes required.